pkg/proxy/context.go

```diff
@@ -0,0 +1,25 @@
+package proxy
+
+import "context"
+
+// private type creates an interface key for Context that cannot be accessed by any other package
+type contextKey int
+
+const (
+	// position counter of the TX in the block
+	contextMmodelName contextKey = iota
+)
+
+// WithModelName sets the model name in the provided context and returns a new context with the updated value.
+// The model name can be retrieved using the contextMmodelName key.
+func WithModelName(ctx context.Context, name string) context.Context {
```

Comment:
In general, I try to avoid passing critical values through in context. I prefer to pass typed arguments to keep things clear and explicit. I would prefer we remove this part of the PR. I think the same functionality of publishing model-specific metrics (very useful) could be accomplished with helper functions called inside of the primary handler function.

Reply:
NP, I will refactor this.  👍 
My motivation to use the request context was to decouple the model name lookup from using the model name. This would provide an extension point via (http) middleware for people to integrate their own model parsing or custom logic when lingo is used as a library. But I realized that the same could be achieved by setting the `X-Model` header  anyway.